examples/v2/chat2.nim

```diff
@@ -171,9 +181,10 @@ proc processInput(rfd: AsyncFD, rng: ref BrHmacDrbgContext) {.async.} =
     let payload = cast[string](message.payload)
     echo &"{payload}"
     info "Hit subscribe handler", topic=topic, payload=payload, contentTopic=message.contentTopic
-  await node.subscribe(topic, handler)
 
-  var chat = Chat(node: node, transp: transp, subscribed: true, connected: false, started: true)
+  # XXX Timing issue with subscribe, need to wait a bit to ensure GRAFT message is sent
```

Comment:
Strange. Are you sure that is what is going wrong here? The subscribe is basically also done already in the `start()` higher up. By the way, can you see the confusion here by allowing both type of subscribes in the public api ?

Reply:
Without this the scenario as tested doesn't work, with it it works. The problem is subscribe being done too early.

What is the confusion you see with subscribe in public API? This is a floodsub vs gossipsub thing, not anything to do with filter subscriptions.